[c10d][doc] add a doc page for NCCL ENVs

shuqiangzhang opened the pull request:
Addressing issue: https://github.com/pytorch/pytorch/issues/128204
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #128235

shuqiangzhang committed:
[c10d][doc] add a doc page for NCCL ENVs

[ghstack-poisoned]

wconstab commented on docs/source/torch_nccl_environment_variables.rst:
```diff
@@ -0,0 +1,33 @@
+.. _torch_nccl_environment_variables:
+
+PYTORCH NCCL Environment Variables
```
Nit: PyTorch ProcessGroupNCCL Environment Variables

wconstab commented on docs/source/torch_nccl_environment_variables.rst:
```diff
@@ -0,0 +1,33 @@
+    - If set to ``1``,enable monitoring thread which aborts the process when the ProcessGroupNCCL Watchdog thread gets stuck and no heartbeat is detected after TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC. This can happen due to calling CUDA/NCCL APIs that may hang. It is Useful to prevent jobs being stuck for a prolonged time than necessary tying up cluster resources.
+  * - ``TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC``
+    - Control the watchdog heartbeat timeout period after which the monitoring thread will abort the process.
+  * - ``TORCH_NCCL_TRACE_BUFFER_SIZE``
```
nit: mention that 0 disables the trace buffer

wconstab commented on docs/source/torch_nccl_environment_variables.rst:
```diff
@@ -0,0 +1,33 @@
+PYTORCH NCCL Environment Variables
+==================================
+.. list-table::
+  :header-rows: 1
```
hmm, do we want to mention default values here? pro: i think its helpful. con: its gonna be out of date. 

Maybe give a pointer instead to the the code where the default values are set?

shuqiangzhang committed:
Update on "[c10d][doc] add a doc page for NCCL ENVs"

Addressing issue: https://github.com/pytorch/pytorch/issues/128204



[ghstack-poisoned]

shuqiangzhang committed:
Update on "[c10d][doc] add a doc page for NCCL ENVs"

Addressing issue: https://github.com/pytorch/pytorch/issues/128204



[ghstack-poisoned]

shuqiangzhang commented:
@pytorchbot merge

shuqiangzhang commented:
@pytorchbot label "release notes: distributed (c10d)"

shuqiangzhang commented:
@pytorchbot merge
